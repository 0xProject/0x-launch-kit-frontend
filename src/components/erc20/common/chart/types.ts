export interface Cord {
    x: number;
    y: number;
}
export enum OrderSide {
    Bid,
    Ask,
}

export interface Margin {
    left: number;
    right: number;
    top: number;
    bottom: number;
}

export interface ChartDimensions {
    height: number;
    margin: Margin;
}
