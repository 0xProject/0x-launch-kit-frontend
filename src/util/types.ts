import { BigNumber, OrderStatus } from '0x.js';
import { SignedOrder } from '@0x/connect';
import { RouterState } from 'connected-react-router';
import { ActionCreator, AnyAction } from 'redux';
import { ThunkAction } from 'redux-thunk';

import { TokenMetaData } from '../common/tokens_meta_data';
import { ExtraArgument } from '../store/index';
import { ThemeModalStyle, ThemeProperties } from '../themes/commons';

export interface TabItem {
    active: boolean;
    onClick: any;
    text: string;
}

export enum Network {
    Mainnet = 1,
    Rinkeby = 4,
    Kovan = 42,
    Ganache = 50,
}

export interface Token {
    address: string;
    decimals: number;
    name: string;
    symbol: string;
    primaryColor: string;
    id?: string;
    icon?: string;
    displayDecimals: number;
}

export interface TokenBalance {
    balance: BigNumber;
    isUnlocked: boolean;
    token: Token;
}

export interface SearchTokenBalanceObject {
    tokenBalances: TokenBalance[];
    tokenToFind: Token | null;
    wethTokenBalance: TokenBalance | null;
}

export enum Web3State {
    Done = 'Done',
    Error = 'Error',
    Loading = 'Loading',
    NotInstalled = 'NotInstalled',
    Locked = 'Locked',
}

export interface BlockchainState {
    readonly ethAccount: string;
    readonly web3State: Web3State;
    readonly tokenBalances: TokenBalance[];
    readonly ethBalance: BigNumber;
    readonly wethTokenBalance: TokenBalance | null;
    readonly gasInfo: GasInfo;
    readonly convertBalanceState: ConvertBalanceState;
}

export interface RelayerState {
    readonly orders: UIOrder[];
    readonly userOrders: UIOrder[];
}

export interface UIState {
    readonly notifications: Notification[];
    readonly fills: Fill[];
    readonly hasUnreadNotifications: boolean;
    readonly stepsModal: StepsModalState;
    readonly orderPriceSelected: BigNumber | null;
}

export interface MarketState {
    readonly currencyPair: CurrencyPair;
    readonly baseToken: Token | null;
    readonly quoteToken: Token | null;
    readonly ethInUsd: BigNumber | null;
    readonly quoteInUsd?: BigNumber | null;
    readonly markets: Market[] | null;
}

export interface StoreState {
    readonly router: RouterState;
    readonly blockchain: BlockchainState;
    readonly relayer: RelayerState;
    readonly ui: UIState;
    readonly market: MarketState;
    readonly collectibles: CollectiblesState;
}

export enum StepKind {
    WrapEth = 'WrapEth',
    ToggleTokenLock = 'ToggleTokenLock',
    BuySellLimit = 'BuySellLimit',
    BuySellMarket = 'BuySellMarket',
    UnlockCollectibles = 'UnlockCollectibles',
    SellCollectible = 'SellCollectible',
    BuyCollectible = 'BuyCollectible',
}

export interface StepWrapEth {
    kind: StepKind.WrapEth;
    currentWethBalance: BigNumber;
    newWethBalance: BigNumber;
    context: 'order' | 'standalone';
}

export interface StepToggleTokenLock {
    kind: StepKind.ToggleTokenLock;
    token: Token;
    isUnlocked: boolean;
    context: 'order' | 'standalone';
}

export interface StepUnlockCollectibles {
    kind: StepKind.UnlockCollectibles;
    collectible: Collectible;
    isUnlocked: boolean;
}

export interface StepBuySellLimitOrder {
    kind: StepKind.BuySellLimit;
    amount: BigNumber;
    price: BigNumber;
    side: OrderSide;
    token: Token;
}

export interface StepBuySellMarket {
    kind: StepKind.BuySellMarket;
    amount: BigNumber;
    side: OrderSide;
    token: Token;
}

export interface StepSellCollectible {
    kind: StepKind.SellCollectible;
    collectible: Collectible;
    startPrice: BigNumber;
    endPrice: BigNumber | null;
    expirationDate: BigNumber;
    side: OrderSide;
}

export interface StepBuyCollectible {
    kind: StepKind.BuyCollectible;
    order: SignedOrder;
    collectible: Collectible;
}

export type Step =
    | StepWrapEth
    | StepToggleTokenLock
    | StepBuySellLimitOrder
    | StepBuySellMarket
    | StepSellCollectible
    | StepBuyCollectible
    | StepUnlockCollectibles;

export interface StepsModalState {
    readonly doneSteps: Step[];
    readonly currentStep: Step | null;
    readonly pendingSteps: Step[];
}

export enum OrderSide {
    Sell,
    Buy,
}

export interface UIOrder {
    rawOrder: SignedOrder;
    side: OrderSide;
    size: BigNumber;
    filled: BigNumber | null;
    price: BigNumber;
    status: OrderStatus | null;
}

export interface OrderBookItem {
    side: OrderSide;
    size: BigNumber;
    price: BigNumber;
}

export interface Spread {
    absolute: BigNumber;
    percentage: BigNumber;
}

export interface OrderBook {
    buyOrders: OrderBookItem[];
    sellOrders: OrderBookItem[];
    mySizeOrders: OrderBookItem[];
}

export interface CurrencyPair {
    base: string;
    quote: string;
}

export interface Market {
    currencyPair: CurrencyPair;
    price: BigNumber | null;
}

export enum NotificationKind {
    CancelOrder = 'CancelOrder',
    Market = 'Market',
    Limit = 'Limit',
    OrderFilled = 'OrderFilled',
}

export interface Fill {
    id: string;
    amountQuote: BigNumber;
    amountBase: BigNumber;
    tokenQuote: Token;
    tokenBase: Token;
    side: OrderSide;
    price: string;
    timestamp: Date;
}

interface BaseNotification {
    id: string;
    kind: NotificationKind;
    timestamp: Date;
}

interface TransactionNotification extends BaseNotification {
    tx: Promise<any>;
}

interface CancelOrderNotification extends TransactionNotification {
    kind: NotificationKind.CancelOrder;
    amount: BigNumber;
    token: Token;
}

interface MarketNotification extends TransactionNotification {
    kind: NotificationKind.Market;
    amount: BigNumber;
    token: Token;
    side: OrderSide;
}

interface LimitNotification extends BaseNotification {
    kind: NotificationKind.Limit;
    amount: BigNumber;
    token: Token;
    side: OrderSide;
}

export interface OrderFilledNotification extends BaseNotification {
    kind: NotificationKind.OrderFilled;
    amount: BigNumber;
    token: Token;
    side: OrderSide;
}

export type Notification = CancelOrderNotification | MarketNotification | LimitNotification | OrderFilledNotification;

export enum OrderType {
    Limit = 'Limit',
    Market = 'Market',
}

export interface GasInfo {
    gasPriceInWei: BigNumber;
    estimatedTimeMs: number;
}

export enum ModalDisplay {
    InstallMetamask = 'INSTALL_METAMASK',
    EnablePermissions = 'ACCEPT_PERMISSIONS',
}

export enum MARKETPLACES {
    ERC20 = 'ERC20',
    ERC721 = 'ERC721',
}

export interface Collectible {
    tokenId: string;
    name: string;
    color: string;
    image: string;
    currentOwner: string;
    assetUrl: string;
    description: string;
    order: SignedOrder | null;
}

export enum AllCollectiblesFetchStatus {
    Request = 'Request',
    Success = 'Success',
}

export enum ConvertBalanceState {
    Failure = 'Failure',
    Request = 'Request',
    Success = 'Success',
}

export interface CollectiblesState {
    readonly allCollectibles: { [tokenId: string]: Collectible };
    readonly allCollectiblesFetchStatus: AllCollectiblesFetchStatus;
    readonly collectibleSelected: Collectible | null;
}

export interface CollectibleMetadataSource {
    fetchAllUserCollectiblesAsync(userAddress: string): Promise<Collectible[]>;
    fetchCollectiblesAsync(tokenIds: string[]): Promise<Collectible[]>;
}

export type ThunkCreator<R = Promise<any>> = ActionCreator<ThunkAction<R, StoreState, ExtraArgument, AnyAction>>;

export enum ButtonVariant {
    Balance = 'balance',
    Buy = 'buy',
    Error = 'error',
    Primary = 'primary',
    Quaternary = 'quaternary',
    Secondary = 'secondary',
    Sell = 'sell',
    Tertiary = 'tertiary',
}

export enum ButtonIcons {
    Warning = 'warning',
}

export interface Filter {
    text: string;
    value: null | string;
}

export interface PartialTheme {
    componentsTheme?: Partial<ThemeProperties>;
    modalTheme?: Partial<ThemeModalStyle>;
}

export interface GeneralConfig {
    title?: string;
    icon?: string;
}

export interface ConfigFile {
    tokens: TokenMetaData[];
    pairs: CurrencyPair[];
    marketFilters?: Filter[];
    theme?: PartialTheme;
    general?: GeneralConfig;
}
