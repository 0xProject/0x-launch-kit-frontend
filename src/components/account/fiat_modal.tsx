import React from 'react';
import Modal from 'react-modal';
import styled from 'styled-components';

import { COINDIRECT_MERCHANT_ID } from '../../common/constants';
import { Theme } from '../../themes/commons';
import { CloseModalButton } from '../common/icons/close_modal_button';
import { IconType, Tooltip } from '../common/tooltip';

interface Props {
    theme: Theme;
    isOpen: boolean;
    ethAccount: string;
    reset: () => void;
}

const ModalContent = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    flex-shrink: 0;
    max-height: 100%;
    min-height: 300px;
    overflow: auto;
    width: 400px;
    height: 610px;
`;

const Title = styled.h1`
    color: ${props => props.theme.componentsTheme.textColorCommon};
    display: flex;
    font-size: 20px;
    font-weight: 600;
    line-height: 1.2;
    margin: 0 0 25px;
    text-align: center;
`;

const TooltipStyled = styled(Tooltip)`
    margin-left: 5px;
`;

export const FiatOnRampModal: React.FC<Props> = props => {
    const { reset, isOpen, theme, ethAccount } = props;
    const fiat_link = `https://business.coindirect.com/buy?merchantId=${COINDIRECT_MERCHANT_ID}&to=eth&address=${ethAccount}`;
    const description = `Disclaimer  <br />
    Veridex now enables easy purchase of Ether using credit & debit cards, through Coindirect! <br />
    Once payment is completed, you can check your payment status on Coindirect and deposit history in your ethereum wallet.<br />
    If you have any questions, please contact: support@coindirect.com`;

    const toolTip = <TooltipStyled description={description} iconType={IconType.Fill} />;
    return (
        <Modal isOpen={isOpen} style={theme.modalTheme}>
            <CloseModalButton onClick={reset} />
            <ModalContent>
                <Title>BUY ETH {toolTip}</Title>
                <iframe title="fiat_on_ramp" src={fiat_link} width="400" height="610" />
            </ModalContent>
        </Modal>
    );
};
