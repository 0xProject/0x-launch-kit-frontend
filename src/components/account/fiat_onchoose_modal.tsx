import React from 'react';
import Modal from 'react-modal';
import { useDispatch, useSelector } from 'react-redux';
import styled, { withTheme } from 'styled-components';

import { openFiatOnRampChooseModal, openFiatOnRampModal, setFiatType } from '../../store/actions';
import { getOpenFiatOnRampChooseModalState } from '../../store/selectors';
import { Theme } from '../../themes/commons';
import { ButtonVariant } from '../../util/types';
import { Button } from '../common/button';
import { CloseModalButton } from '../common/icons/close_modal_button';

interface Props {
    theme: Theme;
}

const ModalContent = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    max-height: 100%;
    overflow: auto;
    width: 310px;
`;

const ModalTitle = styled.h1`
    color: ${props => props.theme.componentsTheme.textColorCommon};
    font-size: 20px;
    font-weight: 600;
    line-height: 1.2;
    margin: 0 0 25px;
    text-align: center;
`;

const ButtonStyled = styled(Button)`
    width: 100%;
    margin: 10px;
`;

const LinkButton = styled.a`
    color: ${props => props.theme.componentsTheme.buttonTextColor};
    text-decoration: none;
`;

const FiatChooseModal: React.FC<Props> = props => {
    const { theme } = props;
    const dispatch = useDispatch();
    const isOpen = useSelector(getOpenFiatOnRampChooseModalState);
    const reset = () => {
        dispatch(openFiatOnRampChooseModal(false));
    };
    /*const chooseApplePay = () => {
        dispatch(setFiatType('APPLE_PAY'));
        dispatch(openFiatOnRampModal(true));
        reset();
    };*/
    const chooseCreditCard = () => {
        dispatch(setFiatType('CREDIT_CARD'));
        dispatch(openFiatOnRampModal(true));
        reset();
    };

    const content = (
        <>
            <ModalTitle>Buy With:</ModalTitle>
            {/*<ButtonStyled onClick={chooseApplePay} variant={ButtonVariant.Portis}>
                <LinkButton>{'Apple Pay'}</LinkButton>
            </ButtonStyled>*/}
            <ButtonStyled onClick={chooseCreditCard} variant={ButtonVariant.Fortmatic}>
                <LinkButton>{'Credit Card'}</LinkButton>
            </ButtonStyled>
        </>
    );

    return (
        <Modal isOpen={isOpen} style={theme.modalTheme}>
            <CloseModalButton onClick={reset} />
            <ModalContent>{content}</ModalContent>
        </Modal>
    );
};

export const FiatChooseModalContainer = withTheme(FiatChooseModal);
