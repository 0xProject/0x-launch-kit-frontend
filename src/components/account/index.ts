export * from './wallet_token_balances';
export * from './wallet_weth_balance';
