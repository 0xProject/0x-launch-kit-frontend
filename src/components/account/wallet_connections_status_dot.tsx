import styled from 'styled-components';

interface StatusProps {
    status?: string;
}

export const WalletConnectionStatusDot = styled.div<StatusProps>`
    background-color: ${props => (props.status ? '#55BC65' : '#ccc')};
    border-radius: 50%;
    height: 10px;
    width: 10px;
`;
