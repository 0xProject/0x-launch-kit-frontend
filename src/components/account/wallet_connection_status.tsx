import React, { HTMLAttributes } from 'react';
import styled from 'styled-components';

import { Dropdown, DropdownPositions } from '../common/dropdown';
import { ChevronDownIcon } from '../common/icons/chevron_down_icon';

import { WalletConnectionStatusDot } from './wallet_connections_status_dot';

const WalletConnectionStatusWrapper = styled.div`
    align-items: center;
    cursor: pointer;
    display: flex;
`;

const WalletConnectionStatusDotStyled = styled(WalletConnectionStatusDot)`
    margin-right: 10px;
`;

const WalletConnectionStatusText = styled.span`
    color: ${props => props.theme.componentsTheme.textColorCommon};
    font-feature-settings: 'calt' 0;
    font-size: 16px;
    font-weight: 500;
    margin-right: 10px;
`;

interface OwnProps extends HTMLAttributes<HTMLSpanElement> {
    walletConnectionContent: React.ReactNode;
    shouldCloseDropdownOnClickOutside?: boolean;
    headerText: string;
    ethAccount: string;
}

type Props = OwnProps;

export class WalletConnectionStatusContainer extends React.PureComponent<Props> {
    public render = () => {
        const {
            headerText,
            walletConnectionContent,
            ethAccount,
            shouldCloseDropdownOnClickOutside,
            ...restProps
        } = this.props;
        const status: string = ethAccount ? 'active' : '';
        const header = (
            <WalletConnectionStatusWrapper>
                <WalletConnectionStatusDotStyled status={status} />
                <WalletConnectionStatusText>{headerText}</WalletConnectionStatusText>
                <ChevronDownIcon />
            </WalletConnectionStatusWrapper>
        );

        const body = <>{walletConnectionContent}</>;
        return (
            <Dropdown
                body={body}
                header={header}
                horizontalPosition={DropdownPositions.Right}
                shouldCloseDropdownOnClickOutside={shouldCloseDropdownOnClickOutside}
                {...restProps}
            />
        );
    };
}
