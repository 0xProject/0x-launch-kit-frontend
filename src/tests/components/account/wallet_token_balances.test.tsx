/**
 * @jest-environment jsdom
 */

import { BigNumber } from '0x.js';
import React from 'react';

import { WalletTokenBalances } from '../../../components/account';
import { TokenBalance, Web3State } from '../../../util/types';
import { mountWithTheme, renderWithTheme } from '../../util/test_with_theme';

const noop = () => ({});
const ZERO = new BigNumber(0);
const tokenDefaults = {
    displayDecimals: 2,
    primaryColor: 'white',
    decimals: 18,
};
const wethTokenBalance = {
    balance: ZERO,
    token: {
        ...tokenDefaults,
        address: '0x100',
        symbol: 'WETH',
        name: 'wETH',
    },
    isUnlocked: true,
};

describe('WalletTokenBalances', () => {
    it('should show one row for each token plus one for the total eth', () => {
        const tokenBalances: TokenBalance[] = [
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x1',
                    symbol: 'MOCK1',
                    name: 'MOCK1',
                },
                isUnlocked: true,
            },
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x2',
                    symbol: 'MOCK2',
                    name: 'MOCK2',
                },
                isUnlocked: true,
            },
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x3',
                    symbol: 'MOCK3',
                    name: 'MOCK3',
                },
                isUnlocked: true,
            },
        ];

        // when
        const wrapper = mountWithTheme(
            <WalletTokenBalances
                ethBalance={ZERO}
                wethTokenBalance={wethTokenBalance}
                tokenBalances={tokenBalances}
                onStartToggleTokenLockSteps={noop}
                web3State={Web3State.Done}
            />,
        );

        // then
        expect(wrapper.find('tbody tr')).toHaveLength(4);
    });

    it('should properly show locked and unlocked tokens', () => {
        const tokenBalances: TokenBalance[] = [
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x1',
                    symbol: 'MOCK1',
                    name: 'MOCK1',
                },
                isUnlocked: true,
            },
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x2',
                    symbol: 'MOCK2',
                    name: 'MOCK2',
                },
                isUnlocked: false,
            },
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x3',
                    symbol: 'MOCK3',
                    name: 'MOCK3',
                },
                isUnlocked: true,
            },
        ];

        const tree = renderWithTheme(
            <WalletTokenBalances
                ethBalance={ZERO}
                wethTokenBalance={wethTokenBalance}
                tokenBalances={tokenBalances}
                onStartToggleTokenLockSteps={noop}
                web3State={Web3State.Done}
            />,
        );

        // when
        expect(tree).toMatchSnapshot();
    });

    it('should call the onToggleTokenLock function when a locked token is clicked', () => {
        const tokenBalances: TokenBalance[] = [
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x1',
                    symbol: 'MOCK1',
                    name: 'MOCK1',
                },
                isUnlocked: true,
            },
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x2',
                    symbol: 'MOCK2',
                    name: 'MOCK2',
                },
                isUnlocked: false,
            },
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x3',
                    symbol: 'MOCK3',
                    name: 'MOCK3',
                },
                isUnlocked: true,
            },
        ];
        const onToggleTokenLock = jest.fn();

        // when
        const wrapper = mountWithTheme(
            <WalletTokenBalances
                ethBalance={ZERO}
                wethTokenBalance={wethTokenBalance}
                tokenBalances={tokenBalances}
                onStartToggleTokenLockSteps={onToggleTokenLock}
                web3State={Web3State.Done}
            />,
        );

        wrapper
            .find('tbody tr')
            .at(2)
            .find('[data-icon="lock"]')
            .simulate('click');

        // then
        expect(onToggleTokenLock).toHaveBeenCalledWith(tokenBalances[1].token, tokenBalances[1].isUnlocked);
    });

    it('should call the onToggleTokenLock function when a unlocked token is clicked', () => {
        const tokenBalances: TokenBalance[] = [
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x1',
                    symbol: 'MOCK1',
                    name: 'MOCK1',
                },
                isUnlocked: true,
            },
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x2',
                    symbol: 'MOCK2',
                    name: 'MOCK2',
                },
                isUnlocked: false,
            },
            {
                balance: new BigNumber(1),
                token: {
                    ...tokenDefaults,
                    address: '0x3',
                    symbol: 'MOCK3',
                    name: 'MOCK3',
                },
                isUnlocked: true,
            },
        ];
        const onToggleTokenLock = jest.fn();

        // when
        const wrapper = mountWithTheme(
            <WalletTokenBalances
                ethBalance={ZERO}
                wethTokenBalance={wethTokenBalance}
                tokenBalances={tokenBalances}
                onStartToggleTokenLockSteps={onToggleTokenLock}
                web3State={Web3State.Done}
            />,
        );
        const rows = wrapper.find('tbody tr');
        rows.at(1)
            .find('[data-icon="lock-open"]')
            .simulate('click');

        // then
        expect(onToggleTokenLock).toHaveBeenCalledWith(tokenBalances[0].token, tokenBalances[0].isUnlocked);
    });
});
