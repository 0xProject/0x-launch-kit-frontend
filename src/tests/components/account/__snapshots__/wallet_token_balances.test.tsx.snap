// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletTokenBalances should properly show locked and unlocked tokens 1`] = `
<div
  className="sc-bwzfXH cfOWoU sc-bdVaJa fJeBhq"
>
  <div
    className="sc-htpNat fMPtTX"
  >
    <h1
      className="sc-bxivhb DvYdZ"
    >
      Token Balances
    </h1>
  </div>
  <div
    className="sc-ifAKCX jlvJWy"
  >
    <table
      className="sc-iwsKbI fpIGxU"
    >
      <thead
        className="sc-gZMcBi bfoAHF"
      >
        <tr
          className="sc-VigVT fIhGog"
        >
          <th
            className="sc-jTzLTM sc-kGXeez bWGQEj"
          >
            Token
          </th>
          <th
            className="sc-jTzLTM sc-kGXeez bWGQEj"
          />
          <th
            className="sc-jTzLTM sc-kGXeez gLdvqw"
          >
            Available Qty.
          </th>
          <th
            className="sc-jTzLTM sc-kGXeez gLdvqw"
          >
            Price (USD)
          </th>
          <th
            className="sc-jTzLTM sc-kGXeez gLdvqw"
          >
            % Change
          </th>
          <th
            className="sc-jTzLTM sc-kgoBCf bLDfeg"
          >
            Locked?
          </th>
        </tr>
      </thead>
      <tbody
        className="sc-hMqMXs eaTBaG"
      >
        <tr
          className="sc-VigVT fIhGog"
        >
          <td
            className="sc-fjdhpX sc-kpOJdX cMhLVX"
          >
            <div
              className="sc-dxgOiQ kjzMiM sc-EHOje dcpbFX"
              color="white"
            >
              <label
                className="sc-bZQynM dstZeq"
              >
                WETH
              </label>
            </div>
          </td>
          <td
            className="sc-fjdhpX sc-ckVGcZ dzaBDi"
          >
            <span
              className="sc-eNQAEJ cOvcCf"
            >
              ETH Total
            </span>
             
             (ETH + wETH)
          </td>
          <td
            className="sc-fjdhpX jsMPdn"
          >
            0.00
          </td>
          <td
            className="sc-fjdhpX jsMPdn"
          >
            -
          </td>
          <td
            className="sc-fjdhpX jsMPdn"
          >
            -
          </td>
          <td
            className="sc-fjdhpX sc-jKJlTe gXnDqt"
          >
            <span
              className="sc-kEYyzF gCLQrS"
              onClick={[Function]}
            >
              <svg
                className="unlockedIcon"
                data-icon="lock-open"
                fill="none"
                height="17"
                viewBox="0 0 13 17"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.54871 4.92014C1.54871 2.21216 3.66224 0 6.24949 0C8.83674 0 10.9503 2.21216 10.9503 4.92014H9.31046C9.31046 3.16567 7.94395 1.71633 6.24949 1.71633C4.55502 1.71633 3.18851 3.16567 3.18851 4.92014V7.33135H9.31046H10.9503C11.8066 7.33135 12.499 8.05602 12.499 8.95232V15.379C12.499 16.2753 11.8066 17 10.9503 17H1.54871C0.692363 17 0 16.2753 0 15.379V8.95232C0 8.05602 0.692363 7.33135 1.54871 7.33135V4.92014Z"
                />
              </svg>
            </span>
          </td>
        </tr>
        <tr
          className="sc-VigVT fIhGog"
        >
          <td
            className="sc-fjdhpX sc-kpOJdX cMhLVX"
          >
            <div
              className="sc-dxgOiQ kjzMiM sc-EHOje dcpbFX"
              color="white"
            >
              <label
                className="sc-bZQynM dstZeq"
              >
                MOCK1
              </label>
            </div>
          </td>
          <td
            className="sc-fjdhpX sc-ckVGcZ dzaBDi"
          >
            <span
              className="sc-eNQAEJ cOvcCf"
            >
              MOCK1
            </span>
             
            - MOCK1
          </td>
          <td
            className="sc-fjdhpX iVqJl"
          >
            0.00
          </td>
          <td
            className="sc-fjdhpX iVqJl"
          >
            -
          </td>
          <td
            className="sc-fjdhpX iVqJl"
          >
            -
          </td>
          <td
            className="sc-fjdhpX sc-jKJlTe gXnDqt"
          >
            <span
              className="sc-kEYyzF gCLQrS"
              onClick={[Function]}
            >
              <svg
                className="unlockedIcon"
                data-icon="lock-open"
                fill="none"
                height="17"
                viewBox="0 0 13 17"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.54871 4.92014C1.54871 2.21216 3.66224 0 6.24949 0C8.83674 0 10.9503 2.21216 10.9503 4.92014H9.31046C9.31046 3.16567 7.94395 1.71633 6.24949 1.71633C4.55502 1.71633 3.18851 3.16567 3.18851 4.92014V7.33135H9.31046H10.9503C11.8066 7.33135 12.499 8.05602 12.499 8.95232V15.379C12.499 16.2753 11.8066 17 10.9503 17H1.54871C0.692363 17 0 16.2753 0 15.379V8.95232C0 8.05602 0.692363 7.33135 1.54871 7.33135V4.92014Z"
                />
              </svg>
            </span>
          </td>
        </tr>
        <tr
          className="sc-VigVT fIhGog"
        >
          <td
            className="sc-fjdhpX sc-kpOJdX cMhLVX"
          >
            <div
              className="sc-dxgOiQ kjzMiM sc-EHOje dcpbFX"
              color="white"
            >
              <label
                className="sc-bZQynM dstZeq"
              >
                MOCK2
              </label>
            </div>
          </td>
          <td
            className="sc-fjdhpX sc-ckVGcZ dzaBDi"
          >
            <span
              className="sc-eNQAEJ cOvcCf"
            >
              MOCK2
            </span>
             
            - MOCK2
          </td>
          <td
            className="sc-fjdhpX iVqJl"
          >
            0.00
          </td>
          <td
            className="sc-fjdhpX iVqJl"
          >
            -
          </td>
          <td
            className="sc-fjdhpX iVqJl"
          >
            -
          </td>
          <td
            className="sc-fjdhpX sc-jKJlTe gXnDqt"
          >
            <span
              className="sc-kEYyzF gCLQrS"
              onClick={[Function]}
            >
              <svg
                className="lockedIcon"
                data-icon="lock"
                fill="none"
                height="16"
                viewBox="0 0 13 16"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6.24949 0C3.66224 0 1.54871 2.21216 1.54871 4.92014V6.33135C0.692363 6.33135 0 7.05602 0 7.95232V14.379C0 15.2753 0.692363 16 1.54871 16H10.9503C11.8066 16 12.499 15.2753 12.499 14.379V7.95232C12.499 7.05602 11.8066 6.33135 10.9503 6.33135V4.92014C10.9503 2.21216 8.83674 0 6.24949 0ZM9.31046 6.33135H3.18851V4.92014C3.18851 3.16567 4.55502 1.71633 6.24949 1.71633C7.94395 1.71633 9.31046 3.16567 9.31046 4.92014V6.33135Z"
                />
              </svg>
            </span>
          </td>
        </tr>
        <tr
          className="sc-VigVT fIhGog"
        >
          <td
            className="sc-fjdhpX sc-kpOJdX cMhLVX"
          >
            <div
              className="sc-dxgOiQ kjzMiM sc-EHOje dcpbFX"
              color="white"
            >
              <label
                className="sc-bZQynM dstZeq"
              >
                MOCK3
              </label>
            </div>
          </td>
          <td
            className="sc-fjdhpX sc-ckVGcZ dzaBDi"
          >
            <span
              className="sc-eNQAEJ cOvcCf"
            >
              MOCK3
            </span>
             
            - MOCK3
          </td>
          <td
            className="sc-fjdhpX iVqJl"
          >
            0.00
          </td>
          <td
            className="sc-fjdhpX iVqJl"
          >
            -
          </td>
          <td
            className="sc-fjdhpX iVqJl"
          >
            -
          </td>
          <td
            className="sc-fjdhpX sc-jKJlTe gXnDqt"
          >
            <span
              className="sc-kEYyzF gCLQrS"
              onClick={[Function]}
            >
              <svg
                className="unlockedIcon"
                data-icon="lock-open"
                fill="none"
                height="17"
                viewBox="0 0 13 17"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.54871 4.92014C1.54871 2.21216 3.66224 0 6.24949 0C8.83674 0 10.9503 2.21216 10.9503 4.92014H9.31046C9.31046 3.16567 7.94395 1.71633 6.24949 1.71633C4.55502 1.71633 3.18851 3.16567 3.18851 4.92014V7.33135H9.31046H10.9503C11.8066 7.33135 12.499 8.05602 12.499 8.95232V15.379C12.499 16.2753 11.8066 17 10.9503 17H1.54871C0.692363 17 0 16.2753 0 15.379V8.95232C0 8.05602 0.692363 7.33135 1.54871 7.33135V4.92014Z"
                />
              </svg>
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
